tests/lax_scipy_test.py

```diff
@@ -125,13 +126,19 @@ def lax_fun(array_to_reduce):
   def testScipySpecialFun(self, scipy_op, lax_op, rng_factory, shapes, dtypes,
                           test_autodiff):
     rng = rng_factory(self.rng())
+    if scipy_op is osp_special.polygamma:
+      dtypes = list(dtypes)
+      dtypes[0] = onp.int32 if dtypes[0] == onp.float32 else onp.int64
```

Comment:
It's a bit unfortunate to hard-code special cases in the test infrastructure... is there a way we could do this by passing a new argument? For example, could we expand `rec.dtypes` to allow a tuple of length nargs such that you can specify different dtypes for each parameter?